src/main/java/com/wanted/babdoduk/restaurant/domain/review/service/ReviewStatService.java

```diff
@@ -0,0 +1,70 @@
+package com.wanted.babdoduk.restaurant.domain.review.service;
+
+import com.wanted.babdoduk.restaurant.domain.restaurant.repository.RestaurantRepository;
+import com.wanted.babdoduk.restaurant.domain.review.dto.RestaurantReviewStatDto;
+import com.wanted.babdoduk.restaurant.domain.review.entity.RestaurantReviewStat;
+import com.wanted.babdoduk.restaurant.domain.review.repository.RestaurantReviewRepository;
+import com.wanted.babdoduk.restaurant.domain.review.repository.RestaurantReviewStatRepository;
+import com.wanted.babdoduk.restaurant.exception.NotFoundRestaurantException;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ReviewStatService {
+
+    private final RestaurantRepository restaurantRepository;
+    private final RestaurantReviewRepository reviewRepository;
+    private final RestaurantReviewStatRepository reviewStatRepository;
+
+    @Transactional
+    public void updateRestaurantReviewStat(Long restaurantId) {
+
+        checkRestaurantExists(restaurantId);
+
+        RestaurantReviewStatDto getReviewStatDto =
+                reviewRepository.getReviewCountAndScoreAverage(restaurantId);
+
+        Optional<RestaurantReviewStat> getRestaurantReviewStat =
+                reviewStatRepository.findByRestaurantId(restaurantId);
+
+        if (getRestaurantReviewStat.isEmpty()) {
```

Comment:
상태가 비어있는 경우 바로 return 을 호출한다면, if-else를 사용하지 않아도 될 것 같아요!

```suggestion
        if (getRestaurantReviewStat.isEmpty()) {
            reviewStatRepository.save(getReviewStatDto.toEntity());
            return;
        }
         getRestaurantReviewStat.get().changeAverageAndCount(getReviewStatDto);
```

Reply:
써주신 코멘트 이해했습니다!
써주신 코멘트에 대해 궁금한 부분이 있어서 여쭤봅니다!
Q. if-else를 사용하는것보다 return을 사용하는 이유가 있을까요? 가독성의 이유일까요?!

Reply:
저는 가독성의 이유가 큽니다.

개인적으로는 코드를 위에서 부터 읽어나갈 때 `if-else` 까지 같이 있으면 else 문으로 들어갈 때 다시 한번 더 조건을 확인하게 되는 경우가 많았습니다.

가령 if 문 만 있을 경우에는 '이런 경우에만 - 이러하다' 라는 느낌이 였다면
if-else 모두 있을 때는 '이런 경우에는 이렇고 조건이 아닌 경우 경우에는 이러하다' 라고 머리 속으로 읽게 되서 한 번 더 조건을 확인하게 되어서 전체적인 코드의 흐름을 파악하기가 어려운 느낌이였습니다.

Reply:
아하!! 답변 감사드립니다! 말씀하신대로 수정하였습니다😄